src/main/java/lotto/domain/calculator/ProfitCalculator.java

```diff
@@ -0,0 +1,29 @@
+package lotto.domain.calculator;
+
+import lotto.domain.Rank;
+
+import java.util.List;
+
+public class ProfitCalculator {
+    public int calculateTotalPrize(List<Rank> ranks) {
+        int totalPrize = 0;
+        for (Rank rank : ranks) {
+            totalPrize += rank.getPrize() * countRanks(ranks, rank);
+        }
+        return totalPrize;
+    }
+
+    private int countRanks(List<Rank> ranks, Rank targetRank) {
+        int count = 0;
+        for (Rank rank : ranks) {
+            if (rank == targetRank) {
+                count++;
+            }
+        }
```

Comment:
```suggestion
        long count = ranks.stream()
                .filter(rank -> rank == targetRank)
                .count();
```
P5: indent의 depth가 2까지 들어가는 부분인지라, 스트림을 사용해서 depth를 0으로 가져가는 방법도 권유드려봅니다!